### DISCUSSION TITLE
Rhino 1.4.0 and renv 1.0.0 issues not activating renv when session restarts

### DISCUSSION THREAD

**Discussion by dawiegriesel** [Appsilon/rhino#482] [General]
Not sure if anyone has experienced these issues since the upgrades.

We have seen quite a few niggling issues:
1. renv::restore() called in the dockerfile tries to call library on box and then the build fails. Fix: downgrade renv to 0.17.3
2. RStudio does not activate renv when you restart the session. Fix: not sure what the problem is.
3. The docker container gives an error: 
Error in eval(quote({ : object '..version..' not found
Calls: source ... local -> eval.parent -> eval -> eval -> eval -> eval
Execution halted

Not sure how to debug this error. 

If anyone has experienced the same issues after upgrading to rhino 1.4.0 please share tips.

Kind regards,
Dawie

**Comment by jakubnowicki** [comment]
Hi @dawiegriesel 
The issue with `renv::restore` was introduced by `renv` 1.0.0. Check this issues for more details: [renv #1611](https://github.com/rstudio/renv/issues/1611) and [box #333](https://github.com/klmr/box/issues/333).
The workaround is to set `options(install.opts = "--no-test-load")`.

**Comment by dawiegriesel** [comment]
Hi @jakubnowicki

Thank you for pointing me in the right direction. Just to add to the workaround when deployed using docker I had to make two additional changes as detailed below. This works and docker builds and runs.
`
RUN sudo -u shiny Rscript -e 'install.packages("withr")'
RUN sudo -u shiny Rscript -e 'options(RENV_CONFIG_SANDBOX_ENABLED = FALSE)'
RUN sudo -u shiny Rscript -e 'withr::with_options(list(install.opts = "--no-test-load"), renv::restore(clean = TRUE, prompt = FALSE))'
`

**Comment by jakubnowicki** [comment]
@dawiegriesel the newest version of `renv` fixed the problem, so your Docker setup should now work without the need to add `no-test-load` option - just upgrade `renv`.
